From 38fde65f6c919350b2bf5d2b61b6e9f783ab2cf7 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Fri, 8 Jul 2011 14:12:53 +1000
Subject: Fix type name and name

Don't overwrite user-specified device name and set the right type_name.
If you're capable of setting the option DeviceName, you might as well set
the Identifier line.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

diff --git a/man/fpit.man b/man/fpit.man
index 097ff33..8bf31f0 100644
--- man/fpit.man
+++ man/fpit.man
@@ -82,10 +82,6 @@ Manipulate the invert and swap options to match screen
 rotations. (Will behave strangely if combined with invert or swap options.)
 New users should consider the \fI"TrackRandR"\fP option instead.
 .TP 4
-.B Option \fI"DeviceName"\fP \fI"name"\fP
-.B Option \fI"DeviceName"\fP \fI"name"\fP
-sets the name of the X device.
-.TP 4
 .B Option \fI"AlwaysCore"\fP \fI"on"\fP
 enables the sharing of the core pointer. When this feature is enabled, the
 device will take control of the core pointer (and thus will emit core events)
diff --git a/src/xf86Fpit.c b/src/xf86Fpit.c
index fa4ace7..e7e4e1a 100644
--- src/xf86Fpit.c
+++ src/xf86Fpit.c
@@ -527,7 +527,6 @@ static int xf86FpitAllocate(InputDriverPtr drv, InputInfoPtr pInfo)
 	priv->fpitOldProximity = 0;
 	priv->fpitIndex = 0;
 	priv->fpitPassive = 0;
-	pInfo->name = XI_TOUCHSCREEN;
 	pInfo->flags = 0 /* XI86_NO_OPEN_ON_INIT */ ;
 	pInfo->device_control = xf86FpitControl;
 	pInfo->read_input = xf86FpitReadInput;
@@ -536,7 +535,7 @@ static int xf86FpitAllocate(InputDriverPtr drv, InputInfoPtr pInfo)
 	pInfo->fd = -1;
 	pInfo->dev = NULL;
 	pInfo->private = priv;
-	pInfo->type_name = "Fujitsu Stylistic";
+	pInfo->type_name = XI_TOUCHSCREEN;
 	return Success;
 }
 
@@ -577,8 +576,6 @@ static int xf86FpitInit(InputDriverPtr drv, InputInfoPtr pInfo, int flags)
 		return BadValue;
 	}
 	priv->fpitDev = strdup(str);
-	pInfo->name = xf86SetStrOption(pInfo->options, "DeviceName", XI_TOUCHSCREEN);
-	xf86Msg(X_CONFIG, "FPIT device name: %s\n", pInfo->name);
 	priv->screen_no = xf86SetIntOption(pInfo->options, "ScreenNo", 0);
 	xf86Msg(X_CONFIG, "Fpit associated screen: %d\n", priv->screen_no);
 	priv->fpitMaxX = xf86SetIntOption(pInfo->options, "MaximumXPosition", FPIT_MAX_X);
-- 
cgit v0.10.2

